engine/derivative.cc

```diff
@@ -639,6 +640,64 @@ namespace MathDAG
        throw error("frac is not differentiable");
   }
   
+  template <>
+  NodePtr SystemOfEquations::derivative<>
+  (const OperationDAG<OperationType::percent>& expr)
+  {
+    if (expr.arguments[0].empty())
+      return zero;
+    else
+      {
+        Expr x(expressionCache, expr.arguments[0][0]);
+        return (100*x)->derivative(*this);
+      }
+  }
+  
+  template <>
+  NodePtr SystemOfEquations::derivative
+  (const OperationDAG<OperationType::gamma>& expr)
+  {
+    if (expr.arguments[0].empty())
+      return zero;
+    else
+      {
+        Expr x(expressionCache, expr.arguments[0][0]);
+        return chainRule(x,polygamma(x,0)*gamma(x)); 
+      }                                                                                       
+  }                                                                                           
+
+  template <>
+  NodePtr SystemOfEquations::derivative
+  (const OperationDAG<OperationType::polygamma>& expr)
+  {
+      assert(expr.arguments.size()==2);
+      if (expr.arguments[0].empty())
+        return zero;
+      else 
+        {
+          Expr x(expressionCache, expressionCache.reverseLookup(*expr.arguments[0][0]));
+          if (expr.arguments[1].empty())
+            return (polygamma(x,1))->derivative(*this);
```

Comment:
Are you sure this works? I would have though chainRule(x, polygamma(x,1)) would be what is needed here. The current expression looks like it would blow up the stack due to recursion.

Reply:
I looks like it works, but I have reverted it to chainRule() which seems to work equally well.